### PR TITLE
Dockerfiles for .NET Core builds

### PR DESCRIPTION
**Pull request by ravimeda**
These Dockerfiles are used to prepare the images needed for .NET Core official builds that are performed in a Docker container.

Fixes https://github.com/dotnet/core-eng/issues/2440

### CODE REVIEW

File: src/build.ps1 [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+ExecuteWithRetry docker pull $ImageBuilderImageName
+$repoRoot = Split-Path -Path "$PSScriptRoot" -Parent
+
+& docker run --rm `
+    -v /var/run/docker.sock:/var/run/docker.sock `
+    -v "${repoRoot}:/repo" `
+    -w /repo/dockerfiles `
+    $ImageBuilderImageName `
+    build --manifest "/repo/dockerfiles/manifest.json" --path "$DockerfilePath" "$ImageBuilderCustomArgs"
```
Comment (ravimeda) [inline src/build.ps1]: `/repo/dockerfiles/manifest.json` - that needs to be updated.

Comment (ravimeda) [reply src/build.ps1]: Replace as -

```powershell
& docker run --rm `
    -v /var/run/docker.sock:/var/run/docker.sock `
    -v "${repoRoot}:/repo" `
    -w /repo/src `
    $ImageBuilderImageName `
    build --manifest "/repo/src/manifest.json" --path "$DockerfilePath" "$ImageBuilderCustomArgs"
```

File: src/build.ps1 [hunk 2]
```diff
@@ -0,0 +1,49 @@
+[cmdletbinding()]
```
Comment (MichaelSimons) [inline src/build.ps1]: I think this should reside at the root of the repo.

File: src/manifest.json [hunk 3]
```diff
@@ -0,0 +1,328 @@
+{
```
Comment (MichaelSimons) [inline src/manifest.json]: I kind of feel like this belongs at the root of the repo as well.

File: src/build.ps1 (hunk 2 above)
Comment (ravimeda) [reply src/build.ps1]: Agree.  Will update.

File: src/manifest.json (hunk 3 above)
Comment (ravimeda) [reply src/manifest.json]: Yes, it would make it consistent with other Docker repos.

**Pull request merged**